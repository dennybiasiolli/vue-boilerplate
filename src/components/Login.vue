<template>
  <div class="Login">
    <h1>Login</h1>
    <p>
      <label>Username {{ username }}</label>
      <br>
      <input
        v-model="username"
        type="text"
        placeholder="Username">
    </p>
    <p>
      <label>Password</label>
      <br>
      <input
        v-model="password"
        type="password"
        placeholder="Password">
    </p>
    <p>
      <label>Remember me</label>
      <br>
      <input
        v-model="remember"
        type="checkbox">
    </p>
    <p>
      <button @click="login">Login</button>
    </p>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'
const { mapGetters, mapActions } = createNamespacedHelpers('auth')

export default {
  data() {
    return {
      username: '',
      password: '',
      remember: true
    }
  },
  computed: Object.assign({}, mapGetters(['isLogged'])),

  methods: Object.assign(
    {
      login() {
        this.loginAsync({
          username: this.username,
          password: this.password,
          remember: this.remember
        }).then(
          () =>
            this.isLogged &&
            this.$router.push(this.$route.query.redirect || '/dashboard')
        )
      }
    },
    mapActions(['loginAsync'])
  )
}
</script>

<style scoped>
.Login {
  align-content: center;
  text-align: center;
}
</style>
