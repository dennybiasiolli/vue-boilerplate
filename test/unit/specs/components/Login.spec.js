import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import Login from '@/components/Login'

const localVue = createLocalVue()
localVue.use(Vuex)

const state = {
  count: 0
}

let getters, actions

// This function creates a new Vuex store
// instance for every new test case.
const createStore = () => {
  getters = {
    isLogged: jest.fn().mockReturnValue(true)
  }
  actions = {
    loginAsync: jest.fn(() => Promise.resolve())
  }
  return new Vuex.Store({
    modules: {
      auth: {
        namespaced: true,
        state,
        actions,
        getters
      }
    }
  })
}

describe('Login', () => {
  let component, vm, element

  beforeEach(() => {
    component = mount(Login, {
      localVue,
      store: createStore(),
      mocks: {
        $router: [],
        $route: {
          query: {
            redirect: undefined
          }
        }
      }
    })
    vm = component.vm
    element = component.element
  })

  test('should match snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof Login.data).toBe('function')
    const defaultData = Login.data()
    expect(defaultData.username).toBe('')
    expect(defaultData.password).toBe('')
    expect(defaultData.remember).toBe(true)
    expect(Login.computed.isLogged).toBeDefined()
    expect(typeof Login.methods.login).toBe('function')
    expect(typeof Login.methods.loginAsync).toBe('function')
  })

  describe('login', () => {
    test('should have been called when clicking on login button', () => {
      jest.spyOn(vm, 'loginAsync').mockReturnValue(Promise.resolve())
      component.find('button').trigger('click')
      expect(vm.loginAsync).toHaveBeenCalledWith({
        username: vm.username,
        password: vm.password,
        remember: vm.remember
      })
      return vm.loginAsync().then(() => {
        expect(vm.$router).toEqual(['/dashboard'])
      })
    })
  })
})
