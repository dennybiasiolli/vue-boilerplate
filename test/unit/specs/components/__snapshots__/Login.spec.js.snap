// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<div
  class="container fluid fill-height"
>
  <div
    class="layout align-center justify-center"
  >
    <div
      class="flex"
    >
      <div
        class="elevation-12 card"
        style="height: auto;"
      >
        <nav
          class="toolbar theme--dark primary"
          style="margin-top: 0px; padding-right: 0px; padding-left: 0px;"
        >
          <div
            class="toolbar__content"
            style="height: 56px;"
          >
            <div
              class="toolbar__title"
            >
              Login
            </div>
          </div>
        </nav>
         
        <div
          class="card__text"
        >
          <form
            novalidate="novalidate"
          >
            <div
              class="input-group input-group--prepend-icon input-group--text-field primary--text"
            >
              <label>
                Username
              </label>
              <div
                class="input-group__input"
              >
                <i
                  aria-hidden="true"
                  class="icon material-icons input-group__prepend-icon"
                >
                  person
                </i>
                <input
                  aria-label="Username"
                  name="login"
                  tabindex="0"
                  type="text"
                />
              </div>
              <div
                class="input-group__details"
              >
                <!---->
              </div>
            </div>
             
            <div
              class="input-group input-group--prepend-icon input-group--text-field primary--text"
            >
              <label>
                Password
              </label>
              <div
                class="input-group__input"
              >
                <i
                  aria-hidden="true"
                  class="icon material-icons input-group__prepend-icon"
                >
                  lock
                </i>
                <input
                  aria-label="Password"
                  name="password"
                  tabindex="0"
                  type="password"
                />
              </div>
              <div
                class="input-group__details"
              >
                <!---->
              </div>
            </div>
             
            <div
              aria-checked="true"
              aria-label="Remember me"
              class="input-group input-group--dirty checkbox input-group--selection-controls input-group--active accent--text"
              role="checkbox"
              tabindex="0"
            >
              <label>
                Remember me
              </label>
              <div
                class="input-group__input"
              >
                <i
                  aria-hidden="true"
                  class="icon icon--selection-control material-icons icon--checkbox"
                >
                  check_box
                </i>
                <div
                  class="input-group--selection-controls__ripple"
                />
              </div>
              <div
                class="input-group__details"
              >
                <!---->
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="card__actions"
        >
          <div
            class="spacer"
          />
           
          <button
            class="btn primary"
            type="button"
          >
            <div
              class="btn__content"
            >
              Login
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
